lib/runtime-core/src/instance.rs

```diff
@@ -595,13 +618,29 @@ fn call_func_with_index(
         Type::I64 => Value::I64(raw as i64),
         Type::F32 => Value::F32(f32::from_bits(raw as u32)),
         Type::F64 => Value::F64(f64::from_bits(raw)),
+        _ => unreachable!(),
     };
 
     match signature.returns() {
         &[] => {
             run_wasm(0 as *mut u64)?;
             Ok(())
         }
+        &[Type::V128] => {
+            let mut result = [0u64; 2];
+
+            run_wasm(result.as_mut_ptr())?;
+
+            let mut bytes = [0u8; 16];
+            let lo = result[0].to_le_bytes();
+            let hi = result[1].to_le_bytes();
+            for i in 0..8 {
+                bytes[i] = lo[i];
+                bytes[i + 8] = hi[i];
+            }
+            rets.push(Value::V128(u128::from_le_bytes(bytes)));
```

Comment:
you can use clone_from_slice like you do in other places; might be worth having a helper function for this

Reply:
I don't see how to use clone_from_slice here? `clone_from_slice` will panic if the sizes are different, which they will be because `lo` is 8 bytes and `bytes` is 16 bytes?